daemon/oci_linux.go

```diff
@@ -912,11 +912,7 @@ func WithDevices(daemon *Daemon, c *container.Container) coci.SpecOpts {
 				devPermissions = append(devPermissions, dPermissions...)
 			}
 
-			var err error
-			devPermissions, err = oci.AppendDevicePermissionsFromCgroupRules(devPermissions, c.HostConfig.DeviceCgroupRules)
-			if err != nil {
-				return err
-			}
+			devPermissions = append(devPermissions, c.HostConfig.DeviceCgroupPermissions...)
```

Comment:
I never thought about that aspect, thanks! I would put forth adding a migration step to avoid either adding more logic to the validation function or adding a second (less strict) one, although I can see why having those kinds of things around could be undesirable. Up to you.